Grammar/python.gram

```diff
@@ -29,7 +29,7 @@ _PyPegen_parse(Parser *p)
 '''
 file[mod_ty]: a=[statements] ENDMARKER { _PyPegen_make_module(p, a) }
 interactive[mod_ty]: a=statement_newline { Interactive(a, p->arena) }
-eval[mod_ty]: a=expressions NEWLINE* ENDMARKER { Expression(a, p->arena) }
+eval[mod_ty]: a=star_expressions NEWLINE* ENDMARKER { Expression(a, p->arena) }
```

Comment:
@lysnikolaou Do you remember why this was `expressions` and not `star_expressions`? I cannot see any edge case that makes this change incorrect but now I am curious.

Reply:
I think that this was for consistency with 3.8. While 3.8 and 3.9 currently fail at parser level for `eval('*a')`, this patch fails at bytecode compiler level.

Reply:
We could make that fail at the parser level if you want but not sure if is worth it.

Reply:
Wouldn't leaving it as is work?

The reason I'm pushing for the error to be in parser level is due to the quality of the error message. Here's an example:

Without this patch:
```
➜  cpython git:(bpo-40631) ✗ cat a.py       
a = [1, 2, 3]
eval('''


*a''')
➜  cpython git:(bpo-40631) ✗ python a.py
Traceback (most recent call last):
  File "/home/lysnikolaou/repos/cpython/a.py", line 2, in <module>
    eval('''
  File "<string>", line 4
    *a
    ^
SyntaxError: invalid syntax
```

With this patch:
```
➜  cpython git:(bpo-40631) ✗ cat a.py
a = [1, 2, 3]
eval('''


*a''')
➜  cpython git:(bpo-40631) ✗ ./python a.py
Traceback (most recent call last):
  File "/home/lysnikolaou/repos/cpython/a.py", line 2, in <module>
    eval('''
  File "<string>", line 4
SyntaxError: can't use starred expression here
```

The first one makes a much better job of making clear what the error is I feel. But if it's too much effort and leaving it as is doesn't work, I'm okay with it.

Reply:
I very much agree. I have closed the PR based on this discussion and Guido's request. Thanks for the examples and the comment.